Lesson7

JuicyFlash opened the pull request:
- Ресурс Вопросов, вложенный в ресурс Тестов
- Контроллер для ресурса вопросов (просмотр, удаление, создание)
- Обработчик исключения для случая когда вопрос не был найден

JuicyFlash committed:
Ресурс Вопросов, вложенный в ресурс Тестов
Контроллер для ресурса вопросов
 - просмотр
 - создание
 - удаление
Обработчик исключений для случая когда вопрос не был найде

JuicyFlash committed:
строка в конце файла

BubuntuClu commented on app/controllers/questions_controller.rb:
```diff
@@ -0,0 +1,41 @@
+  end
+
+  def create
+    Question.create(test_id: @test.id, body: params[:question][:body])
```
Параметры с фронта могут быть не валидные, это надо учитывать
```
question = @test.questions.new(questions_params)
if question.save
  Render good
else
  Render bad
end
```

BubuntuClu commented on app/controllers/questions_controller.rb:
```diff
@@ -0,0 +1,41 @@
+
+  rescue_from ActiveRecord::RecordNotFound, with: :rescue_with_questions_not_found
+
+  before_action :find_test, only: %i[index create]
```
ты находишь тест для экшена `create`, но объект теста там не используешь. зачем тогда его находить?

BubuntuClu commented on app/controllers/questions_controller.rb:
```diff
@@ -0,0 +1,41 @@
+  # Удаление вопроса
+  # /questions/:id
+  def destroy
+    Question.find(params[:id]).destroy
```
этот код дублируется в нескольких методах. надо вынести в колбэк поиск объекта

BubuntuClu commented on app/views/questions/index.html.erb:
```diff
@@ -0,0 +1,17 @@
+    <th>Вопросы</th>
+
+  </tr>
+  <% @test.questions.each do |qst| %>
```
ты не ограничен в месте, не стоит сокращать названия `qst`. пиши подробные имена, чтобы не надо было догадываться до их смысла

JuicyFlash committed:
Сокращённое название переменной qst изменено на question

JuicyFlash committed:
Добавлен колбэк find_question
Добавлена проверка валидности в create
Добавлен метод параметров question_params

JuicyFlash commented:
-Переменная qst переименована в question
-Добавлен коллбэк find_question, для методов destroy и show
-Добавлена проверка валидности в методе create (коллбэк для create сохранился, тк объект test в нём используется)
-Добавлен метод параметров question_params
